Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -376,36 +376,47 @@ void CDMInstanceOpenCDM::removeSessionData(const String& sessionId, LicenseType,
     if (session->remove(responseMessage)) {
         RefPtr<SharedBuffer> cleanMessage = cleanResponseMessage(responseMessage);
         if (cleanMessage) {
-            GST_TRACE("message length %u", cleanMessage->size());
             SharedBuffer& initData = session->initData();
             keys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{initData, MediaKeyStatus::Released});
+            GST_DEBUG("session %s removed, message length %u", sessionId.utf8().data(), cleanMessage->size());
             callback(WTFMove(keys), cleanMessage.releaseNonNull(), SuccessValue::Succeeded);
         } else {
-            GST_ERROR("OpenCDM::update claimed to have a message, but it was not correctly formatted");
+            GST_WARNING("message of size %u incorrectly formatted as session %s removal answer", responseMessage.size(), sessionId.utf8().data());
             callback(WTFMove(keys), std::nullopt, SuccessValue::Failed);
         }
     } else {
         SharedBuffer& initData(session->initData());
         MediaKeyStatus keyStatus = mediaKeyStatusFromOpenCDM(responseMessage);
         keys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{initData, keyStatus});
+        GST_WARNING("could not remove session %s", sessionId.utf8().data());
         callback(WTFMove(keys), std::nullopt, SuccessValue::Failed);
     }
 
-    if (!removeSession(sessionId))
-        GST_WARNING("Failed to remove session %s", sessionId.utf8().data());
+#ifndef NDEBUG
+    bool result =
+#endif
+        removeSession(sessionId);
+    ASSERT(result);
 }
 
 void CDMInstanceOpenCDM::closeSession(const String& sessionId, CloseSessionCallback callback)
 {
     auto session = lookupSession(sessionId);
     if (!session) {
-        GST_WARNING("cannot close session %s because it does not exist", sessionId.utf8().data());
+        GST_WARNING("cannot close non-existing session %s", sessionId.utf8().data());
         return;
     }
-    session->close();
-
-    if (!removeSession(sessionId))
-        GST_WARNING("Failed to remove session %s", sessionId.utf8().data());
+#ifndef NDEBUG
```

Comment:
All these NDEBUG stuff looks like it would case a compile error, and the surrounding  formatting looks a bit suspicious.